lib/mirage.ml

```diff
@@ -917,6 +917,69 @@ let ethernet = Type ETHERNET
 let etif network =
   impl ethernet network (module Ethif)
 
+module Arpv4 = struct
+  type t = {
+    ethernet: ethernet impl;
+    clock: clock impl;
+    time: time impl;
+  } (* is the type checker not inferring that this is an arp impl? I think this
+    is the root problem.  not sure how to tell it that?  how does it work for
+       other modules? *)
```

Comment:
to answer the question in the comment: what you are defining here is how to configure an ARP implementation, ie. what parameters needs to be given to (i) the functor and (ii) the constructor function (ie, `create` or `connect` or ...)